src/models/Question.ts

```diff
@@ -0,0 +1,80 @@
+import { MyContext } from "../context";
+import { MySqlModel } from "./MySqlModel";
+
+export class Question extends MySqlModel {
+  public templateId: number;
+  public sectionId: number;
+  public sourceQuestionId?: number;
+  public questionTypeId: number;
+  public questionText: string;
+  public requirementText?: string;
+  public guidanceText?: string;
+  public sampleText?: string;
+  public required: boolean;
+  public displayOrder: number;
+  public isDirty: boolean;
+
+  private tableName = 'questions';
+
+  constructor(options) {
+    super(options.id, options.created, options.createdById, options.modified, options.modifiedById);
+
+    this.templateId = options.templateId;
+    this.sectionId = options.sectionId;
+    this.sourceQuestionId = options.sourceQuestionId;
+    this.questionTypeId = options.questionTypeId;
+    this.questionText = options.questionText;
+    this.requirementText = options.requirementText;
+    this.guidanceText = options.guidanceText;
+    this.sampleText = options.sampleText;
+    this.required = options.required || false;
+    this.displayOrder = options.displayOrder;
+    this.isDirty = options.isDirty || false;
+  }
+
+  // Validation to be used prior to saving the record
+  async isValid(): Promise<boolean> {
```

Comment:
Add a isValid() function so we can check for the required fields for Question class.